extensions/jms/src/main/java/org/mule/extensions/jms/api/operation/JmsRequestReply.java

```diff
@@ -0,0 +1,278 @@
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.UseConfig;
+import org.mule.runtime.extension.api.annotation.param.display.Summary;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import javax.jms.DeliveryMode;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.Topic;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Operation that allows the user to send a message to a JMS {@link Destination} and waits for a response
+ * either to the provided {@code ReplyTo} destination or to a temporary {@link Destination} created dynamically
+ * @since 4.0
+ */
+public class JmsRequestReply {
+
+  private static final Logger logger = LoggerFactory.getLogger(JmsRequestReply.class);
+  private JmsResultFactory messageFactory = new JmsResultFactory();
+
+  /**
+   * Operation that allows the user to send a message to a JMS {@link Destination} and waits for a response
+   * either to the provided {@code ReplyTo} destination or to a temporary {@link Destination} created dynamically
+   *
+   * @param config the current {@link JmsProducerConfig}
+   * @param connection the current {@link JmsConnection}
+   * @param destination the name of the {@link Destination} where the {@link Message} should be sent
+   * @param isTopic {@code true} if the {@link Destination} is a {@link Topic}
+   * @param messageBuilder the {@link MessageBuilder} used to create the {@link Message} to be sent
+   * @param messageBuilder the {@link MessageBuilder} used to create the {@link Message} to be sent
+   * @param persistentDelivery {@code true} if {@link DeliveryMode#PERSISTENT} should be used
+   * @param priority the {@link Message#getJMSPriority} to be set
+   * @param timeToLive the time the message will be in the broker before it expires and is discarded
+   * @param timeToLiveUnit unit to be used in the timeToLive configurations
+   * @param ackMode the {@link AckMode} that will be configured over the Message and Session
+   * @param deliveryDelay Only used by JMS 2.0. Sets the delivery delay to be applied in order to postpone the Message delivery
+   * @param deliveryDelayUnit Time unit to be used in the deliveryDelay configurations
+   * @return a {@link Result} with the reply {@link Message} content as {@link Result#getOutput} and its properties
+   * and headers as {@link Result#getAttributes}
+   * @throws JmsExtensionException if an error occurs
+   */
+  @OutputResolver(output = JmsOutputResolver.class)
+  public Result<Object, JmsAttributes> requestReply(@UseConfig JmsProducerConfig config, @Connection JmsConnection connection,
```

Comment:
boxed types without default provides a way to do `resolveOverride` with the config elements